docs/gen-ai/Package Structure.md

```diff
@@ -0,0 +1,7 @@
+The GenAI project will be a collection of popular open source AI models. It will be organized in the following structure:
```

Comment:
Who would we direct to use this package vs the ONNX runtime GenAI package? e.g. in what situation would someone be better off using this one instead of that one? Models that it doesn't support that this one does?  Or is this more for experimentation / samples rather than us actually publishing a nuget package we'd encourage folks to adopt? (Which would be fine, just trying to understand the landscape.)

Reply:
> Who would we direct to use this package vs the ONNX runtime GenAI package?

ORT GenAI package is what we'd recommend if all you need is inferencing. As far as I know though there are no plans at this time to enable fine-tuning in that library. While the scenario this package currently enables is inferencing just like ORT GenAI, having a version of these models in TorchSharp would enable fine-tuning scenarios as well. Currently in ML.NET, for [object detection](https://devblogs.microsoft.com/dotnet/object-detection-ml-dotnet-model-builder/#object-detection-in-model-builder) and [language scenarios](https://devblogs.microsoft.com/dotnet/introducing-the-ml-dotnet-text-classification-api-preview/), TorchSharp is what powers the fine-tuning of those models. As a next step for this package, we'd be looking to enable fine-tuning. 

> Models that it doesn't support that this one does

In terms of model coverage there is overlap. However, ORT Gen AI provides access to other language models that this library doesn't.

![image](https://github.com/dotnet/machinelearning/assets/46974588/7ef24fde-d159-4229-a5be-75d5281c5912)

> experimentation / samples rather than us actually publishing a nuget package we'd encourage folks to adopt?

Initially this would be an experimental package as we get feedback and enable fine-tuning capabilities.

Reply:
They target for different model format. This GenAI packages loads the same weight as pytorch model. (.safetensor or .bin) while onnx runtime GenAI loads .onnx model format, which usually requires an extra converting step.

So if a model provides both onnx or .safetensor model weight format, then it's up to you to either use this GenAI package or the onnx runtime GenAI package. In most cases however, you need to convert .safetensor or .bin to .onnx by yourself or download 3p onnx model weight from enthusiastic community members before using ONNX runtime GenAI package.